src/nextjournal/clerk/viewer.cljc

```diff
@@ -1032,17 +1035,86 @@
+                     elision-content-length
+                     (contains? #{:map-entry} name)
+                     (reduce + 1 (map #(:content-length % 0) value))
+                     (vector? value)
+                     (->> value
+                       (map #(:content-length % 0))
+                       (reduce + (+ (count opening-paren) (count closing-paren)))
+                       (+ (dec (count value))))
+                     :else 0)
+                   :type name))
+        x))
+   wrapped-value))
+
+(defn compute-expanded-at [{:as state :keys [indents expanded-at prev-type]}
+                           {:nextjournal/keys [value]
+                            :keys [content-length path type]
+                            :or {content-length 0}}]
+  (let [max-length (- 80 (reduce + 0 indents))
+        expanded? (< max-length content-length)
+        state' (assoc state :expanded-at (assoc expanded-at path expanded?)
+                            #_(if expanded?
+                              (assoc expanded-at path true)
+                              expanded-at)
+                            :prev-type type
+                            :indents (conj
+                                       (->> indents (take (count path)) vec)
+                                       (cond
+                                         (contains? #{:map-entry} prev-type) (or content-length 0)
+                                         (vector? value) 2
+                                         :else 1)))]
+    (if (vector? value)
+      (reduce compute-expanded-at state' value)
+      state')))
+
+(defn assign-expanded-at [wrapped-value]
+  (cond-> wrapped-value
+    (:content-length wrapped-value) (assoc :nextjournal/expanded-at
+                                           (:expanded-at (compute-expanded-at {:indents [] :expanded-at {}} wrapped-value)))))
+
+(comment
+  (-> (compute-expanded-at
+        {:indents [] :expanded-at {}}
+        (present {:a-vector [1 2 3] :a-list '(123 234 345) :a-set #{1 2 3 4}}))
+    :expanded-at
+    keys
+    sort)
+  (= (count "[1 2 [1 [2] 3] 4 5]")
+     (:content-length (assign-content-lengths (present [1 2 [1 [2] 3] 4 5]))))
+  (= (count "{:a-vector [1 2 3] :a-list (123 234 345) :a-set #{1 2 3 4}}")
+     (:content-length (assign-content-lengths (present {:a-vector [1 2 3] :a-list '(123 234 345) :a-set #{1 2 3 4}}))))
+  ;; Check for elisions as well
+  (assign-content-lengths (present {:foo (vec (repeat 2 {:baz (range 30) :fooze (range 40)})) :bar (range 20)})))
+
 (defn present
   "Returns a subset of a given `value`."
   ([x] (present x {}))
-  ([x opts]
+  ([x {:as opts :keys [auto-expand-results?]}]
    (-> (ensure-wrapped-with-viewers x)
        (merge {:!budget (atom (:budget opts 200))
                :path (:path opts [])
                :current-path (:current-path opts [])}
               opts)
        present*
-       assign-closing-parens)))
+       assign-closing-parens
+       (cond->
```

Comment:
Would it make more sense to do this at call site in `->result`, instead of for each call of `present`?